app/safe/internal/server/handle/handle.go

```diff
@@ -11,20 +11,48 @@
 package handle
 
 import (
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
 	"github.com/vmware-tanzu/secrets-manager/app/safe/internal/server/route"
 	"github.com/vmware-tanzu/secrets-manager/core/crypto"
 	"github.com/vmware-tanzu/secrets-manager/core/log"
+	"github.com/vmware-tanzu/secrets-manager/core/validation"
 	"io"
 	"net/http"
 )
 
-func InitializeRoutes() {
+// InitializeRoutes initializes the HTTP routes for the web server. It sets up an
+// HTTP handler function for the root URL ("/"). The handler uses the given
+// X509Source to retrieve X.509 SVIDs for validating incoming connections.
+//
+// Parameters:
+//   - source: A pointer to a `workloadapi.X509Source`, used to obtain X.509 SVIDs.
+//
+// Note: The InitializeRoutes function should be called only once, usually
+// during server initialization.
+func InitializeRoutes(source *workloadapi.X509Source) {
```

Comment:
Passing the `source` here so that we can validate the SVID of VSecM Safe at every API call. — Before this, we were doing it only during bootstrapping.